sormas-backend/src/main/java/de/symeda/sormas/backend/labmessage/LabMessageFacadeEjb.java

```diff
@@ -179,13 +179,31 @@ public LabMessageDto toDto(LabMessage source) {
 			target.setTestReports(source.getTestReports().stream().map(t -> TestReportFacadeEjb.toDto(t)).collect(toList()));
 		}
 		target.setReportId(source.getReportId());
+		if (source.getSample() != null) {
+			target.setSample(source.getSample().toReference());
+		}
 
 		return target;
 	}
 
 	@Override
-	public LabMessageDto getByUuid(String uuid) {
-		return toDto(labMessageService.getByUuid(uuid));
+	public LabMessageDto getByUuid(String uuid, boolean getIfDeleted) {
+		if (getIfDeleted) {
```

Comment:
Okay, so I shall remove the getIfDeleted parameter again. Do you want the LabMessageFacadeEjb.getByUuid(...) to return deleted lab messages? If so, I'm a little worried that developers will assume a different behaviour or completely overlook that lab messages are a CoreADO.

Reply:
Yes, I would say to remove it. It will just work the same as all other CoreADO entities so it would be consistent with other facades.

Maybe it would be better to read the labmessage by uuid and throw an exception if it is marked as deleted so the developers would be forced to handle it so it wouldn't happen that developers "overlook that lab messages are a CoreADO" and also there won't be NullpointerExceptions because of deleted entries.

If you want to keep this parameter I would suggest to refactor a but the method and kepp only the critarieFilter approach but add the "not deleted" filter based on the that, this would it won't look like 2 methonds in one. But anyway the return value of the method should be checked against null because excluding deleted entries increases the possibility of null pointer exceptions.